phac-nml/irida-next
# Email notifications for user access when adding/removing direct members to/from a group/project

## What does this PR do and why?
Implemented emails for notifying the user and managers when a direct member has been added/removed to/from a group/project.
Addresses another third of #476.

Note: A manager is a member that has a role of owner or maintainer.

## Screenshots or screen recordings
 Access **granted** email being sent to a **user**:
![image](https://github.com/phac-nml/irida-next/assets/325703/5ca97e46-a5f6-416a-a7a9-8c64adadca65)
Access **granted** email being sent to a **manager**:
![image](https://github.com/phac-nml/irida-next/assets/325703/190ba6ee-6109-49b7-8540-e95d53708882)Access **revoked** email being sent to a **user**:
![image](https://github.com/phac-nml/irida-next/assets/325703/e67226ed-9198-4b25-9275-eaec68ef9dbb)Access **revoked** email being sent to a **manager**:
![image](https://github.com/phac-nml/irida-next/assets/325703/6b1d1592-2005-445f-aa4e-e0258d147e76)

## How to set up and validate locally
1. Install local mock email service, [Mailhog](https://github.com/mailhog/MailHog).
1. After setup, open a terminal and run `~/go/bin/MailHog`.
1. Run irida-next.
1. Login to irida-next.
1. Navigate to a `group` or `project` that the logged in user is a manager of. Alternatively, you could create a new `group` or `project`.
1. Select the `Members` page.
1. Try adding a member that doesn't already have access. (Verify inherited & group link access.)
1. Try removing a member that only has direct access.
1. Check mock email service for sent emails. Open a browser and navigate to http://0.0.0.0:8025/.

Note: Email previews are located at http://localhost:3000/rails/mailers/member_mailer.
## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
